boot/bootutil/include/bootutil/bootutil_public.h

```diff
@@ -47,6 +47,14 @@
 extern "C" {
 #endif
 
+#ifndef ALIGN_UP
+#define ALIGN_UP(num, align)    (((num) + ((align) - 1)) & ~((align) - 1))
```

Comment:
Iooks like we need something even more common header for both xxx_public.h files

Reply:
Indeed. `bootutil_public.c` and `bootutil_misc.c` also duplicate several identical helpers functions.
I believe this refactor may deserve a dedicated PR due to the amount of changes, as it would make the current one even more extensive.

Reply:
^^ sounds good to me. Let's not complicate this PR more.